azurerm/resource_arm_front_door.go

```diff
@@ -0,0 +1,1459 @@
+	if len(input) == 0 {
+		return &[]frontdoor.BackendPool{}
+	}
+
+	output := make([]frontdoor.BackendPool, 0)
+
+	for _, bp := range input {
+		backendPool := bp.(map[string]interface{})
+
+		backendPoolName := backendPool["name"].(string)
+		backendPoolLoadBalancingName := backendPool["load_balancing_name"].(string)
+		backendPoolHealthProbeName := backendPool["health_probe_name"].(string)
+
+		backends := backendPool["backend"].([]interface{})
+
+		result := frontdoor.BackendPool{
+			ID:   utils.String(helper.GetFrontDoorSubResourceId(subscriptionId, resourceGroup, frontDoorName, "BackendPools", backendPoolName)),
+			Name: utils.String(backendPoolName),
+			BackendPoolProperties: &frontdoor.BackendPoolProperties{
+				// ResourceState
+				Backends:              expandArmFrontDoorBackend(backends),
+				LoadBalancingSettings: expandArmFrontDoorSubResource(subscriptionId, resourceGroup, frontDoorName, "LoadBalancingSettings", backendPoolLoadBalancingName),
+				HealthProbeSettings:   expandArmFrontDoorSubResource(subscriptionId, resourceGroup, frontDoorName, "HealthProbeSettings", backendPoolHealthProbeName),
+			},
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorBackend(input []interface{}) *[]frontdoor.Backend {
+	if len(input) == 0 {
+		return &[]frontdoor.Backend{}
+	}
+
+	output := make([]frontdoor.Backend, 0)
+
+	for _, be := range input {
+		backend := be.(map[string]interface{})
+
+		address := backend["address"].(string)
+		hostHeader := backend["host_header"].(string)
+		enabled := backend["enabled"].(bool)
+		httpPort := int32(backend["http_port"].(int))
+		httpsPort := int32(backend["https_port"].(int))
+		priority := int32(backend["priority"].(int))
+		weight := int32(backend["weight"].(int))
+
+		result := frontdoor.Backend{
+			Address:           utils.String(address),
+			BackendHostHeader: utils.String(hostHeader),
+			EnabledState:      expandArmFrontDoorBackendEnabledState(enabled),
+			HTTPPort:          utils.Int32(httpPort),
+			HTTPSPort:         utils.Int32(httpsPort),
+			Priority:          utils.Int32(priority),
+			Weight:            utils.Int32(weight),
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorBackendEnabledState(isEnabled bool) frontdoor.BackendEnabledState {
+	if isEnabled {
+		return frontdoor.Enabled
+	}
+
+	return frontdoor.Disabled
+}
+
+func expandArmFrontDoorBackendPoolsSettings(enforceCertificateNameCheck bool) *frontdoor.BackendPoolsSettings {
+	enforceCheck := frontdoor.EnforceCertificateNameCheckEnabledStateDisabled
+
+	if enforceCertificateNameCheck {
+		enforceCheck = frontdoor.EnforceCertificateNameCheckEnabledStateEnabled
+	}
+
+	result := frontdoor.BackendPoolsSettings{
+		EnforceCertificateNameCheck: enforceCheck,
+	}
+
+	return &result
+}
+
+func expandArmFrontDoorFrontendEndpoint(input []interface{}, subscriptionId string, resourceGroup string, frontDoorName string) *[]frontdoor.FrontendEndpoint {
```

Comment:
Would it make sense to pregenerate the front door ID and pass it in:
```suggestion
func expandArmFrontDoorFrontendEndpoint(input []interface{}, frontDoorID string) *[]frontdoor.FrontendEndpoint {
```

Reply:
Fixed.